communication/mavsdkvehicleserver.cpp

```diff
@@ -21,10 +21,11 @@ MavsdkVehicleServer::MavsdkVehicleServer(QSharedPointer<VehicleState> vehicleSta
     std::shared_ptr<mavsdk::ServerComponent> serverComponent = mMavsdk.server_component_by_type(mavsdk::Mavsdk::ServerComponentType::Autopilot);
 
     // Create server plugins
+    MavlinkParameterServer::createSingleton(serverComponent);
```

Comment:
I think it would be better to create the singleton in the vehicle's main, not to save the pointer to the singleton object in MavlinkParameterServer, but do a nullptr check when a parameter update is attempted. The user of the ParameterServer gets a little inconvenience (initialising the singleton) but I think this feature should be "enabled" explicitly in main anyways.
In case the singleton is created here, there is an implicit dependency in what order objects can be created in main: MavsdkVehicleServer needs to be created before any class using parameters. Further, MavsdkvehicleServer needs VehicleState created before -> VehicleState cannot provide any parameters but it might be something we want to do (?).